src/server/stream_family.cc

```diff
@@ -1319,6 +1323,97 @@ OpResult<uint32_t> OpAck(const OpArgs& op_args, string_view key, string_view gna
   return acknowledged;
 }
 
+OpResult<ClaimInfo> OpAutoClaim(const OpArgs& op_args, string_view key, const ClaimOpts& opts) {
+  OpResult<pair<stream*, streamCG*>> cgr_res = FindGroup(op_args, key, opts.group);
+  if (!cgr_res)
+    return cgr_res.status();
+  stream* stream = cgr_res->first;
+  streamCG* group = cgr_res->second;
+
+  if (stream == nullptr || group == nullptr) {
+    return OpStatus::KEY_NOTFOUND;
+  }
+
+  streamConsumer* consumer = nullptr;
+  int64_t attempts = opts.count * 10;
+
+  unsigned char start_key[sizeof(streamID)];
```

Comment:
Since this it used quite a lot in this file, how about adding something like:
```cpp
using SerializedStreamId = std::array<uint8_t, sizeof(streamID)>;
```
And use that?

(of course my naming might not be ideal)

Reply:
i felt like keeping the low level array as it will be interacting with the c functions such as the `streamEncodeID` which takes void*... it is a bit more convenient...

Reply:
yes, by easier I meant I don't have to do .data() call...